java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/persistence/AdhocPersistence.java

```diff
@@ -15,6 +15,12 @@
   private final Function<String, Optional<T>> persistenceLoad;
   private final BiConsumer<String, T> persistenceStore;
 
+  /**
+   * Creates a new {@code AdhocPersistence} instance. This is an implementation of {@link Persistence}.
```

Comment:
can add some docs in here to specify that the `Function<String, Optional<T>>` and the `BiConsumer<String, T>` needs to be implemented separately in order to use this.